src/language-js/printer-estree.js

```diff
@@ -4111,6 +4111,10 @@ function printTypeScriptModifiers(path, options, print) {
   return concat([join(" ", path.map(print, "modifiers")), " "]);
 }
 
+function getTypeParametersGroupId(node, options) {
+  return Symbol.for("prettier.typeParameters" + options.locStart(node));
+}
```

Comment:
This mean to use different Symbol for different Node?

This work?

```js
const typeParametersGroupIds = new WeakMap();
function getTypeParametersGroupId(node, options) {
  if (!typeParametersGroupIds.has(node)) {
   typeParametersGroupIds.set(node, Symbol());
  }
  return typeParametersGroupIds.get(node);
}
```

Reply:
Yes, your suggestion looks good. Thank you.